### PR TITLE
supporting indices predictions in topk filter function

### PR DESCRIPTION
**Pull request by msxingpeng**
This PR is support the indices predictions input in topk filtering function

### CODE REVIEW

**Review by shonohs — APPROVED** [review]

File: vision_evaluation/prediction_filters.py [hunk 1]
```diff
@@ -18,33 +18,40 @@ def identifier(self):
...
 
-    def __init__(self, k: int):
+    def __init__(self, k: int, prediction_mode='prob'):
         """
         Args:
             k: k predictions with highest confidence
         """
         assert k >= 0
-
+        self.prediction_mode = prediction_mode
```
[the remark this answers is not in the document]
Comment (msxingpeng) [reply vision_evaluation/prediction_filters.py:30]: Done

File: vision_evaluation/prediction_filters.py [hunk 2]
```diff
@@ -18,33 +18,40 @@ def identifier(self):
 
 class TopKPredictionFilter(PredictionFilter):
 
-    def __init__(self, k: int):
+    def __init__(self, k: int, prediction_mode='prob'):
```
[the remark this answers is not in the document]
Comment (msxingpeng) [reply vision_evaluation/prediction_filters.py:21]: Done

File: vision_evaluation/prediction_filters.py [hunk 3]
```diff
@@ -18,33 +18,42 @@ def identifier(self):
...
+            if k == 0:
+                top_k_pred_indices = np.array([[] for i in range(predictions.shape[1])], dtype=int)
+            elif k == 1:
+                top_k_pred_indices = np.argmax(predictions, axis=1)
+                top_k_pred_indices = top_k_pred_indices.reshape((-1, 1))
+            else:
+                top_k_pred_indices = np.argpartition(predictions, -k, axis=1)[:, -k:]
         else:
-            top_k_pred_indices = np.argpartition(predictions, -k, axis=1)[:, -k:]
+            top_k_pred_indices = predictions[:, :k]
```
Comment (pj-ms) [inline vision_evaluation/prediction_filters.py:56]: What happens when k > M? 

Can we add a unit test for this filter? Thanks

Comment (msxingpeng) [reply vision_evaluation/prediction_filters.py:56]: k>M has already be considered in https://github.com/microsoft/vision-evaluation/blob/0f34dcbc7ebd037357698808d056f0dba442f03e/vision_evaluation/prediction_filters.py#L40

The logic is the same with the case when the prediction is prob. The dimension of prob vector C can also be smaller than k.

Comment (msxingpeng) [reply vision_evaluation/prediction_filters.py:56]: Unit test has already been added in `test_evaluator.py`

File: test/test_evaluators.py [hunk 4]
```diff
@@ -187,38 +187,46 @@ class TestMultilabelClassificationEvaluator(unittest.TestCase):
...
         ks = [0, 1, 2, 3]
         expectations = [0, 0.61111, 0.88888, 1.0]
         for i in range(len(ks)):
             recall_eval = RecallEvaluator(TopKPredictionFilter(ks[i]))
-            recall_eval.add_predictions(self.PREDICTIONS, self.TARGETS)
+            recall_eval.add_predictions(self.PROB_PREDICTIONS, self.TARGETS)
+            self.assertAlmostEqual(recall_eval.get_report(average='samples')[f"recall_top{ks[i]}"], expectations[i], places=4)
+
+        for i in range(len(ks)):
+            recall_eval = RecallEvaluator(TopKPredictionFilter(ks[i], prediction_mode='indices'))
```
Comment (msxingpeng) [inline test/test_evaluators.py:228]: here is the unit test for TopKPredictionFilter when `prediction_mode` is `indices` @pj-ms

**Review by pj-ms — APPROVED** [review]

**Pull request merged**